src/test/java/com/sggc/controllers/SggcControllerIT.java

```diff
@@ -0,0 +1,220 @@
+package com.sggc.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sggc.errors.ApiError;
+import com.sggc.exceptions.TooFewSteamIdsException;
+import com.sggc.exceptions.UserHasNoGamesException;
+import com.sggc.exceptions.VanityUrlResolutionException;
+import com.sggc.models.sggc.SggcResponse;
+import com.sggc.models.steam.request.GetCommonGamesRequest;
+import com.sggc.services.GameService;
+import com.sggc.services.UserService;
+import com.sggc.services.VanityUrlService;
+import com.sggc.validation.ValidationResult;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import static com.sggc.services.VanityUrlService.VANITY_URL_NOT_ALPHANUMERIC_ERROR_MESSAGE;
```

Comment:
Don't rely on constants in the application code. If they don't match the requirements the test won't catch this.